packages/tunnel-client/src/lib/tunnel.ts

```diff
@@ -266,53 +228,63 @@ export class Tunnel extends (EventEmitter as new () => TypedEmitter<TunnelEvents
     http2Connections,
   }: Omit<TunnelInfo, "multiplexingRemotePort"> & {
     multiplexingRemotePort: number;
-  }): Promise<TunnelHTTP2Cluster> {
+  }): Promise<void> {
     const localProtocol = localHttps ? "https" : "http";
     this.logger.debug(
-      "establishing tunnel %s://%s:%s <> %s:%s",
+      "establishing tunnel %s://%s:%s <> %s:%s using cluster workers",
       localProtocol,
       localHost,
       localPort,
       remoteHost,
       multiplexingRemotePort,
     );
 
-    const commonTunnelOpts = {
-      logger: this.logger,
-      localHost,
-      localPort,
-      localHttps,
-      allowInvalidCert,
-      proxyAllUrls,
-      rewriteHostnameToAppUrl,
-      enableDnsCache,
-    };
+    const numWorkers = http2Connections || DEFAULT_HTTP2_NUMBER_OF_CONNECTIONS;
+    const workerPath = path.resolve(__dirname, "./worker-entrypoint.js");
 
-    const numConnections =
-      http2Connections ?? DEFAULT_HTTP2_NUMBER_OF_CONNECTIONS;
-    const sockets = await Promise.all(
-      Array.from({ length: numConnections }).map(() =>
-        this.openSocket({
-          useTls,
-          remoteHost,
-          multiplexingRemotePort,
-          tunnelPassphrase,
-          sendAuthOkAck: true,
-        }),
-      ),
-    );
+    cluster.setupPrimary({
+      exec: workerPath,
+      silent: false,
+      execArgv: ["--max-old-space-size=2048"],
```

Comment:
Can we make this dynamic? We might overcommit on smaller CI runners which might be bad. Also do we need 2G of heap space? I wonder what the default on CI runners is 🤔

Reply:
Let's just drop this actually and let Node use it's default. I don't think it's needed.